api/controllers/PackageController.js

```diff
@@ -105,13 +107,87 @@ module.exports = {
       sort = Utils.parseSort(req),
       criteria = Utils.parseCriteria(req);
 
-    Package.findAll({
-      where: criteria,
-      limit: limit,
-      offset: offset,
-      order: sort,
-      include: []
-    }).then(function(packages) {
+    var packageVersionCriteria = {};
+    if(criteria.parser_version)
+      packageVersionCriteria.parser_version = criteria.parser_version;
+    var packageCriteria = criteria;
+    delete packageCriteria.parser_version;
+
+    var queryPromise;
+
+    var isPopularitySort = (sort === 'popularity');
+    if( Object.prototype.toString.call(sort ) === '[object Array]'){
```

Comment:
Can't you use `https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/isArray` to test if it's an array ?